File: Packages/CaptureKit/Sources/CaptureKit/FileNamer.swift
import Foundation

public enum FileNamer {
    public static func fileName(for date: Date, ext: String, prefix: String = "Screenshot",
                                timeZone: TimeZone = .current) -> String {
        let f = DateFormatter()
        f.locale = Locale(identifier: "en_US_POSIX")
        f.timeZone = timeZone
        f.dateFormat = "yyyy-MM-dd 'at' HH.mm.ss"
        return "\(prefix) \(f.string(from: date)).\(ext)"
    }
}

File: windows/src/BetterScreenshot.Capture/FileNamer.cs
using System.Globalization;

namespace BetterScreenshot.Capture;

/// <summary>
/// Deterministic, locale-independent capture filenames, e.g. "Screenshot 2026-06-02 at 14.32.10.png".
/// Uses dots (not colons) for the time so the name is valid on Windows and matches the macOS app.
/// </summary>
public static class FileNamer
{
    public static string Name(DateTime date, string ext, string prefix = "Screenshot")
    {
        string stamp = date.ToString("yyyy-MM-dd 'at' HH.mm.ss", CultureInfo.InvariantCulture);
        return $"{prefix} {stamp}.{ext}";
    }
}
